WireDomain/Sources/WireDomain/Repositories/TeamRepository.swift

```diff
@@ -72,6 +83,29 @@ final class TeamRepository: TeamRepositoryProtocol {
         await storeTeamLocally(team)
     }
 
+    func deleteTeamMember(
+        time: Date,
+        userID: UUID,
+        teamID: UUID
+    ) async throws {
+        let user = try await userRepository.fetchUser(with: userID)
+
+        try await context.perform { [context] in
+            guard let member = user.membership else {
+                throw TeamRepositoryError.userNotAMemberInTeam(user: userID, team: teamID)
+            }
+
+            if user.isSelfUser {
+                let notification = AccountDeletedNotification(context: context)
+                notification.post(in: context.notificationContext)
```

Comment:
I know it's like this in current code, but if we delete the account, is it necessary to delete the member. We're going to logout and teardown the whole data, no?